src/tests/ie_test_utils/common_test_utils/data_utils.hpp

```diff
@@ -253,6 +254,77 @@ void inline fill_random_unique_sequence(T* rawBlobDataPtr,
     std::copy(elems.begin(), elems.end(), rawBlobDataPtr);
 }
 
+/** @brief Fill tensor with random data.
+ *
+ * @param tensor Target tensor
+ * @param range Values range
+ * @param start_from Value from which range should start
+ * @param k Resolution of floating point numbers.
+ * - With k = 1 every random number will be basically integer number.
+ * - With k = 2 numbers resolution will 1/2 so outputs only .0 or .50
+ * - With k = 4 numbers resolution will 1/4 so outputs only .0 .25 .50 0.75 and etc.
+ */
+template<ov::element::Type_t DT>
+void inline fill_tensor_random(ov::Tensor& tensor, const uint32_t range = 10, int32_t start_from = 0,
+                               const int32_t k = 1, const int seed = 1) {
+    using T = typename ov::element_type_traits<DT>::value_type;
+    auto *rawBlobDataPtr = static_cast<T*>(tensor.data());
+    if (DT == ov::element::u4 || DT == ov::element::i4 ||
+        DT == ov::element::u1) {
+        fill_data_random(rawBlobDataPtr, tensor.get_byte_size(), range, start_from, k, seed);
+    } else {
+        fill_data_random(rawBlobDataPtr, tensor.get_size(), range, start_from, k, seed);
+    }
+}
+
+template<ov::element::Type_t DT>
+void inline
+fill_tensor_random_float(ov::Tensor& tensor, const uint32_t range, int32_t start_from, const int32_t k,
+                         const int seed = 1) {
+    using T = typename ov::element_type_traits<DT>::value_type;
+    std::default_random_engine random(seed);
+    // 1/k is the resolution of the floating point numbers
+    std::uniform_int_distribution<int32_t> distribution(k * start_from, k * (start_from + range));
+
+    auto *rawBlobDataPtr = static_cast<T*>(tensor.data());
+    for (size_t i = 0; i < tensor.get_size(); i++) {
+        auto value = static_cast<float>(distribution(random));
+        value /= static_cast<float>(k);
+        if (DT == ov::element::Type_t::f16) {
+            rawBlobDataPtr[i] = static_cast<T>(ngraph::float16(value).to_bits());
+        } else if (DT == ov::element::Type_t::bf16) {
+            rawBlobDataPtr[i] = static_cast<T>(ngraph::bfloat16(value).to_bits());
+        } else {
+            rawBlobDataPtr[i] = static_cast<T>(value);
+        }
+    }
+}
+
+template<>
+void inline fill_tensor_random<ov::element::f32>(ov::Tensor& tensor,
+                                                 const uint32_t range,
+                                                 int32_t start_from,
+                                                 const int32_t k,
+                                                 const int seed) {
+    fill_tensor_random_float<ov::element::f32>(tensor, range, start_from, k, seed);
+}
+
+template<>
+void inline fill_tensor_random<ov::element::f16>(ov::Tensor& tensor,
```

Comment:
why we cannot use directly template function?

Reply:
Floating values requires slightly different approach for generation, and again, the behavior aligned with `fill_blob()` functions